tensorflow_serving/tools/docker/Dockerfile.devel

```diff
@@ -51,6 +51,7 @@ RUN curl -fSsL -O https://bootstrap.pypa.io/get-pip.py && \
     rm get-pip.py
 
 RUN pip --no-cache-dir install \
+    future>=0.17.1 \
```

Comment:
is this specific min version required?

Reply:
It is otherwise we get the issue that @karthikvadla reported here #1363 while building `master`.

Reply:
the error shown here: https://github.com/tensorflow/serving/issues/1363#issue-450014405 is due to the version string not being well formed. no? can you point me the error that necessitates use of explicit version here (sorry I might have missed it).

Reply:
Oh I see.
Actually the error is:
```
DEBUG: Rule 'io_bazel_rules_docker' indicated that a canonical reproducible form can be obtained by modifying arguments shallow_since = "1556410077 -0400"
Loading: 0 packages loaded
Analyzing: target //tensorflow/tools/pip_package:build_pip_package (1 packages loaded, 0 targets configured)
Analyzing: target //tensorflow/tools/pip_package:build_pip_package (3 packages loaded, 0 targets configured)
Analyzing: target //tensorflow/tools/pip_package:build_pip_package (73 packages loaded, 30 targets configured)
Analyzing: target //tensorflow/tools/pip_package:build_pip_package (295 packages loaded, 6382 targets configured)
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/external/org_tensorflow/tensorflow/tools/git/gen_git_source.py", line 29, in <module>
    from builtins import bytes  # pylint: disable=redefined-builtin
ImportError: No module named builtins
ERROR: /tensorflow/tensorflow/core/BUILD:2778:1: no such package '@local_config_git//': Traceback (most recent call last):
	File "/tensorflow/third_party/git/git_configure.bzl", line 61
		_fail(result.stderr)
	File "/tensorflow/third_party/git/git_configure.bzl", line 14, in _fail
		fail(("%sGit Configuration Error:%s %...)))
Git Configuration Error: Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/external/org_tensorflow/tensorflow/tools/git/gen_git_source.py", line 29, in <module>
    from builtins import bytes  # pylint: disable=redefined-builtin
ImportError: No module named builtins

 and referenced by '//tensorflow/core:version_info_gen'
ERROR: /tensorflow/tensorflow/core/BUILD:2778:1: no such package '@local_config_git//': Traceback (most recent call last):
	File "/tensorflow/third_party/git/git_configure.bzl", line 61
		_fail(result.stderr)
	File "/tensorflow/third_party/git/git_configure.bzl", line 14, in _fail
		fail(("%sGit Configuration Error:%s %...)))
Git Configuration Error: Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/external/org_tensorflow/tensorflow/tools/git/gen_git_source.py", line 29, in <module>
    from builtins import bytes  # pylint: disable=redefined-builtin
ImportError: No module named builtins

 and referenced by '//tensorflow/core:version_info_gen'
ERROR: /tensorflow/tensorflow/core/BUILD:2778:1: no such package '@local_config_git//': Traceback (most recent call last):
	File "/tensorflow/third_party/git/git_configure.bzl", line 61
		_fail(result.stderr)
	File "/tensorflow/third_party/git/git_configure.bzl", line 14, in _fail
		fail(("%sGit Configuration Error:%s %...)))
Git Configuration Error: Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/external/org_tensorflow/tensorflow/tools/git/gen_git_source.py", line 29, in <module>
    from builtins import bytes  # pylint: disable=redefined-builtin
ImportError: No module named builtins

 and referenced by '//tensorflow/core:version_info_gen'
ERROR: Analysis of target '//tensorflow/tools/pip_package:build_pip_package' failed; build aborted: no such package '@local_config_git//': Traceback (most recent call last):
	File "/tensorflow/third_party/git/git_configure.bzl", line 61
		_fail(result.stderr)
	File "/tensorflow/third_party/git/git_configure.bzl", line 14, in _fail
		fail(("%sGit Configuration Error:%s %...)))
Git Configuration Error: Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/68a62076e91007a7908bc42a32e4cff9/external/org_tensorflow/tensorflow/tools/git/gen_git_source.py", line 29, in <module>
    from builtins import bytes  # pylint: disable=redefined-builtin
ImportError: No module named builtins
```

This happens while building `mkl` development container from source.

Reply:
yeah your patch fixes above `builtins`import error, but still i see parse issue when release branch `r1.14` is used.
```
docker build     --build-arg TF_SERVING_VERSION_GIT_BRANCH="r1.14"     --build-arg HTTP_PROXY=${HTTP_PROXY}     --build-arg HTTPS_PROXY=${HTTPS_PROXY}     --build-arg http_proxy=${http_proxy}     --build-arg https_proxy=${https_proxy}     -f Dockerfile.devel-mkl -t tensorflow/serving:latest-devel-mkl .
```

```
Processing /tmp/pip/tensorflow_serving_api-1.14.0rc0-py2.py3-none-any.whl
ERROR: Exception:
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/cli/base_command.py", line 178, in main
    status = self.run(options, args)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/commands/install.py", line 352, in run
    resolver.resolve(requirement_set)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 131, in resolve
    self._resolve_one(requirement_set, req)
  File "/usr/local/lib/python2.7/dist-packages/pip/_internal/resolve.py", line 345, in _resolve_one
    set(req_to_install.extras) - set(dist.extras)
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 2969, in extras
    return [dep for dep in self._dep_map if dep]
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3014, in _dep_map
    self.__dep_map = self._compute_dependencies()
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3024, in _compute_dependencies
    reqs.extend(parse_requirements(req))
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3085, in parse_requirements
    yield Requirement(line)
  File "/usr/local/lib/python2.7/dist-packages/pip/_vendor/pkg_resources/__init__.py", line 3094, in __init__
    raise RequirementParseError(str(e))
RequirementParseError: Parse error at "'(===1.14'": Expected stringEnd
The command '/bin/sh -c bazel build --color=yes --curses=yes     ${TF_SERVING_BAZEL_OPTIONS}     --verbose_failures     --output_filter=DONT_MATCH_ANYTHING     ${TF_SERVING_BUILD_OPTIONS}     tensorflow_serving/tools/pip_package:build_pip_package &&     bazel-bin/tensorflow_serving/tools/pip_package/build_pip_package     /tmp/pip &&     pip --no-cache-dir install --upgrade /tmp/pip/tensorflow_serving*.whl &&     rm -rf /tmp/pip' returned a non-zero code: 2
```
So, this is not fix for issue i reported

Reply:
@ashahba  can you please update your description with actual error and remove reference to the issue.